src/pal/src/thread/process.cpp

```diff
@@ -61,6 +61,14 @@ SET_DEFAULT_DEBUG_CHANNEL(PROCESS); // some headers have code with asserts, so d
 #include <stdint.h>
 #include <dlfcn.h>
 
+#ifdef __linux__
+#include <sys/syscall.h> // __NR_membarrier
+// Ensure __NR_membarrier is defined for x64 portable build.
```

Comment:
Why just `amd64`? The portable builds are meant to work everywhere, not just on x64.

Reply:
Ok, I see - the syscall ids are different between architectures

Reply:
I was actually thinking that we would add it for all the four architectures we support. But let me check what is our build system for portable builds for ARM and ARM64, maybe that's not necessary for the others.